packages/@aws-cdk/aws-apigatewayv2/README.md

```diff
@@ -19,6 +19,100 @@
 
 This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
 
+
+## Introduction
+
+Amazon API Gateway is an AWS service for creating, publishing, maintaining, monitoring, and securing REST, HTTP, and WebSocket APIs at any scale. API developers can create APIs that access AWS or other web services, as well as data stored in the AWS Cloud. As an API Gateway API developer, you can create APIs for use in your own client applications. Or you can make your APIs available to third-party app developers. For more infomation, read the [Amazon API Gateway Developer Guide](https://docs.aws.amazon.com/apigateway/latest/developerguide/welcome.html).
+
+
+## API
+
+This construct library at this moment implements API resources from [AWS::ApiGatewayV2::Api](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-api.html), which supports both `WebSocket APIs` and `HTTP APIs`.
+
+For more information about WebSocket APIs, see [About WebSocket APIs in API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-overview.html) in the _API Gateway Developer Guide_. For more information about HTTP APIs, see [HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api.html) in the _API Gateway Developer Guide_.
+
+To create [REST APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-rest-api.html), use `@aws-cdk/aws-apigateway` instead.
+
+
+### HTTP API
+
+`CfnApi` is the L1 construct to create either `WebSocket APIs` or `HTTP APIs`. At this moment, `HTTP APIs` supports both `Lambda Proxy Integration` and `HTTP Proxy Integration`. See [Working with AWS Lambda Proxy Integrations for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html) and [Working with HTTP Proxy Integrations for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-http.html) for more information.
+
+
+To create `HTTP APIs` with `Lambda Proxy Integration`, simply use  `LambdaProxyApi`.
+
+
+```ts
+// Create a HTTP API with Lambda Proxy Integration as its $default route
+const api = new apigatewayv2.LambdaProxyApi(stack, 'LambdaProxyApi', {
+  handler
+});
+```
+
+To create `HTTP APIs` with `HTTP Proxy Integration`, use `HttpProxyApi` instead.
+
+```ts
+// Create a HTTP API with HTTP Proxy Integration
+new apigatewayv2.HttpProxyApi(stack, 'HttpProxyApi', {
+  url: 'https://aws.amazon.com'
+});
+```
+
```

Comment:
We should also add an example of how to create an HTTP API without Lambda proxy or HTTP proxy, i.e., a regular HTTP API on which routes can be attached.